pkg/cmd/repo/setdefault/setdefault.go

```diff
@@ -96,6 +88,14 @@ func NewCmdSetDefault(f *cmdutil.Factory, runF func(*SetDefaultOptions) error) *
 				return errors.New("must be run from inside a git repository")
 			}
 
+			if len(args) > 0 {
+				var err error
+				opts.Repo, err = parseRepo(args[0], opts)
+				if err != nil {
+					return err
+				}
+			}
+
```

Comment:
**Is there a reason this logic needed to be moved to later within `RunE`?  Does this create a problem above because `opts.Repo` is unset?**

Depending on the motivation for moving the logic, this might have to be undone as non-interactive usage will break because `opts.Repo` is unset.

https://github.com/cli/cli/blob/b4b8e8a13bb99bae479929048ea5f64808d0bc32/pkg/cmd/repo/setdefault/setdefault.go#L80-L83